### PR TITLE
Inject siteId

### PR DESCRIPTION
**Pull request by momo-ozawa**
## Description
This PR injects the site id into MobilePayKit.

## Notes
Apply patch to test the new createOrders endpoint (D65254-code)

For sandbox info, follow instructions in these posts:
- pbMoDN-mH-p2
- pbArwn-24D-p2

### Running tests
1. Run MobilePayKitTests test target
2. ✅ Tests should pass

### Running the app

1. Configure the bearer token and site id in `ProductListViewModel`

https://github.com/Automattic/MobilePay-Apple/blob/03400b306c01ab8b1108302885577f5fca252157/Example/iOS/MobilePayKit-iOS/ProductListViewModel.swift#L31-L35

2. Run the iOS app
3. Purchase the product by tapping on it
4. Tap Confirm
5. Tap OK
6. On Xcode debugger menu, click on the Manage StoreKit Transactions menu item 
7. ✅ There should be a record of a product being purchased (See example below)

Screen Shot 2021-07-19 at 16 32 32

### CODE REVIEW

File: Source/MobilePayKit.swift [hunk 1]
```diff
@@ -4,10 +4,12 @@ import StoreKit
 public struct MobilePayKitConfiguration {
     public let oAuthToken: String
     public let bundleId: String?
+    public let siteId: String
 
-    public init(oAuthToken: String, bundleId: String?) {
+    public init(oAuthToken: String, bundleId: String?, siteId: String) {
```
Comment (leandroalonso) [inline Source/MobilePayKit.swift]: `siteId` is optional, I think we can make it optional and assign a default nil value, wdyt?

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@momo-ozawa works as described!

I'd just change `siteId` to be optional, given it's not a mandatory field.

File: Source/MobilePayKit.swift (hunk 1 above)
Comment (momo-ozawa) [reply Source/MobilePayKit.swift]: Fixed! 033e9f5

Comment (leandroalonso) [reply Source/MobilePayKit.swift]: There's a build error after this change, you need to update the remaining places too. :D

Comment (momo-ozawa) [reply Source/MobilePayKit.swift]: Oof! Fixed... updated `site_id` to `Int?` as well, per D65254-code

**Review by leandroalonso — APPROVED** [review]
Backend is currently failing if no `site_id` is given, this will be addressed in a diff.

**Pull request merged**